src/logcollector/read_journald.c

```diff
@@ -0,0 +1,64 @@
+#include "shared.h"
+#include "logcollector.h"
+#include <systemd/sd-journal.h>
+
+void prime_sd_journal(sd_journal **jrn) {
+  int ret;
+  ret = sd_journal_open(&jrn, SD_JOURNAL_LOCAL_ONLY);
+  ret = sd_journal_get_fd(jrn);
+  ret = sd_journal_seek_tail(jrn);
+  // prime sd_journal_next before the reader loop
+  ret = sd_journal_previous(jrn);
```

Comment:
`ret` should be checked for failure.

Reply:
Some of these will also complain about the pointer. They expect `*` but get `**`.